src/Examine.Test/Examine.Lucene/Search/FluentApiTests.cs

```diff
@@ -117,21 +207,121 @@ public void Uppercase_Category()
 
-                var results = query.Execute();
+
+                var results = query.WithFacets(facets => facets.Facet("nodeName")).Execute();
+
+                var facetResults = results.GetFacet("nodeName");
+
+                Assert.AreEqual(3, results.TotalItemCount);
+                Assert.AreEqual(3, facetResults.Count());
+            }
+        }
+
+        [Test]
+        public void FacetsConfig_SetIndexName_Long()
+        {
+            var fieldDefinitionCollection = new FieldDefinitionCollection(new FieldDefinition("parentID", FieldDefinitionTypes.Integer), new FieldDefinition("LongValue", FieldDefinitionTypes.FacetLong));
+
+            var facetsConfig = new FacetsConfig();
+            facetsConfig.SetIndexFieldName("LongValue", "facet_longvalue");
+
+            var analyzer = new StandardAnalyzer(LuceneInfo.CurrentVersion);
+            using (var luceneDir = new RandomIdRAMDirectory())
+            using (var indexer = GetTestIndex(
+                luceneDir,
+                analyzer,
+                fieldDefinitionCollection,
+                facetsConfig: facetsConfig))
+            {
+                indexer.IndexItems(new[] {
+                    ValueSet.FromObject(1.ToString(), "cOntent",
+                        new { nodeName = "location 1", bodyText = "Zanzibar is in Africa", LongValue = 10L }),
+                    ValueSet.FromObject(2.ToString(), "cOntent",
+                        new { nodeName = "location 2", bodyText = "In Canada there is a town called Sydney in Nova Scotia", LongValue = 20L }),
+                    ValueSet.FromObject(3.ToString(), "cOntent",
+                        new { nodeName = "location 3", bodyText = "Sydney is the capital of NSW in Australia", LongValue = 30L })
+                    });
+
+                var searcher = indexer.Searcher;
+
+                var query = searcher.CreateQuery("cOntent").All();
+
+                Console.WriteLine(query);
+
+
+                var results = query.WithFacets(facets => facets.Facet("LongValue", new Int64Range[]
+                {
+                    new Int64Range("10", 10, true, 11, true),
+                    new Int64Range("20", 20, true, 21, true),
+                    new Int64Range("30", 30, true, 31, true),
+                })).Execute();
+
+                var facetResults = results.GetFacet("LongValue");
 
                 Assert.AreEqual(3, results.TotalItemCount);
+                Assert.AreEqual(3, facetResults.Count());
             }
         }
 
         [Test]
-        public void NativeQuery_Phrase()
+        public void FacetsConfig_SetIndexName_Double()
         {
+            var fieldDefinitionCollection = new FieldDefinitionCollection(new FieldDefinition("parentID", FieldDefinitionTypes.Integer), new FieldDefinition("DoubleValue", FieldDefinitionTypes.FacetDouble));
+
+            var facetsConfig = new FacetsConfig();
+            facetsConfig.SetIndexFieldName("DoubleValue", "facet_doublevalue");
```

Comment:
This test scenario should not occur. SetIndexFieldName is only meant to be called when there is a separate Taxonomy index in use. See the Lucene.NET docs for this method.